modules/filesystem/bm.nu

```diff
@@ -0,0 +1,83 @@
+# simple bookmark module
+
+# List all bookmarked paths
+export def list [] {
+  let bm_path = (get_path)
+
+  if (not ($bm_path | path exists))  {
+    [{name: "prev", path: ("~/" | path expand) }] | save $bm_path
+  }
+    open ($bm_path)
+}
+
+def get_path [] {
+  $env.XDG_DATA_HOME | 
+  path join "bookmarks.nuon"
+}
+
+# Reset the bookmarks
+export def reset [] {
+  list |
+  where name == "prev" |
+  save -f (get_path)
+}
+
+# Add a new bookmark with an optional name
+export def add [
+  pth: path       # Path to bookmark to.
```

Comment:
i would not expand the path here :thinking: 
because, for instance, when i run `bm` for the first time i get
```
[[name, path]; [prev, /home/amtoine]]
```
in the `bm.nuon` file

maybe having `~` instead of the expanded home might be better to allow sharing / versioning this bookmark file? :yum:

Reply:
I expanded it, because i remembered for my scripts the paths didn't work for `~` . Has that been fixed?
Then i'll just change it to `~` :smile:

Reply:
not sure what you mean by "paths didn't work for `~`" but you could just `str replace $env.HOME '~'` for instance

Reply:
that's not a big deal :relieved:

Reply:
for instance, `bm list | update path { str replace $env.HOME '~' }` does the job :+1:

Reply:
really it's up to you :wink: 
if the intent is to share and version control this bookmark file, then i'd remove the very personal "home" part.
otherwise, it's not a big deal at all :smirk:

Reply:
> for instance, `bm list | update path { str replace $env.HOME '~' }` does the job +1

I never thought of using that :open_mouth: . I will add some commits to use `~` in the file.

Reply:
If we're going to require a BM_PATH then some doc comments would be helpful. Typically Windows users won't have it, nor XDG_anything, nor HOME.

Reply:
agreed :+1:

Reply:
i like to put the in a `main` command
```
export def main [] { help bm }
```
and add information about environment variables in there :yum:

Reply:
e.g. https://github.com/amtoine/nupm/blob/main/mod.nu#L305-L317

Reply:
I think the help looks great.

It won't work for Windows users unless the custom env var is set. I'm not sure how it will fail on the first run since none of the 3 env vars you're checking for exist on Windows. 

If we want it to be cross platform, we'll need to check for $env.USERPROFILE, which is the Windows equivalent of $env.HOME.

Reply:
Added Windows path version. I haven't tested it, but i think it should work. The script also checks if the dir in question exists or not, and creates it appropretly.

Reply:
Thanks. I'm good with landing this then.